x-pack/plugin/ent-search/qa/rest/roles.yml

```diff
@@ -2,6 +2,6 @@ entsearch:
   cluster:
     - manage
   indices:
-    - names: '*'
-      privileges:
-        - manage
+    - names: [ "*" ]
+      allow_restricted_indices: true
```

Comment:
Why is this needed (allow_restricted_indices)?

Reply:
I see, that's because we use the wrong `indices()` in the get engine request. It should be the engine alias name and not the system index name as I outlined in my comment below.

Reply:
It looks like it's still required in order for the tests to work. 

Presumably, this is because the system index is protected and the alias is set on the protected system index when it's created? I wasn't able to see concretely where in the code this would be enforced though. Do you have other ideas?

Reply:
This is because this PR uses `ENGINE_ALIAS_NAME` for the `indices` in the get request. It should use `Engine.getEngineAliasName(engineName)` instead. We don't want to use the system index alias name, this index is a protected system index that only the engine origin can access. What we expect here is that the security checks the permission on the alias name of the newly created index.

Reply:
🤦‍♀️ Thanks, fixing.